### PR TITLE
Replace Array[File] with File inputs in single-sample pipeline

### PR DESCRIPTION
**Pull request by epiercehoffman**
### Updates
* As a workaround for the Terra character/array length limit for input JSON and data table TSV uploads, convert Array[File] inputs in single-sample pipeline to File inputs containing a list of reference panel paths, one per line.
  * Automatically generate and upload these list files during `create_test_batch.py` when a GCS destination is provided with `--file-list-bucket`
  * Update `ref_panel_1kg.json` with file lists for current paths in the JSON
* Various updates to single-sample Terra configurations to make it work with latest code (this branch)
  * Add path to single-sample Terra JSON to Terra validation script to make it easier to maintain
* Update ref panel `clean_vcf` with a new one generated by running the latest version of `MakeCohortVcf` including fixing end tags and rescaling GQ

### Testing
* Ran single-sample pipeline without MELT in Terra successfully with this branch and latest dockers
* Ran single-sample pipeline with MELT on cromwell - successful until unrelated bug in MakeCohortVcf (deemed it would take too long to re-run and the sections with changes already succeeded, as well as the run on Terra)
* Validated all WDLs and JSONs with womtool & Terra validation script

### CODE REVIEW

File: inputs/values/ref_panel_1kg.json [hunk 1]
```diff
@@ -315,6 +315,7 @@
         "gs://gatk-sv-ref-panel-1kg/outputs/GATKSVPipelineBatch/38c65ca4-2a07-4805-86b6-214696075fef/call-GatherSampleEvidenceBatch/GatherSampleEvidenceBatch/bb5e70f1-7186-4e29-96f8-dcb9ee000642/call-GatherSampleEvidence/shard-154/GatherSampleEvidence/f5159902-35ad-4e8e-ac76-2e6278ac387f/call-PESRCollection/PESRCollection/4538b819-c0fe-4d69-bcc1-1658a293e6de/call-RunPESRCollection/cacheCopy/NA21122.disc.txt.gz.tbi",
         "gs://gatk-sv-ref-panel-1kg/outputs/GATKSVPipelineBatch/38c65ca4-2a07-4805-86b6-214696075fef/call-GatherSampleEvidenceBatch/GatherSampleEvidenceBatch/bb5e70f1-7186-4e29-96f8-dcb9ee000642/call-GatherSampleEvidence/shard-155/GatherSampleEvidence/9dfac5cf-9a0b-4070-ad5b-2fe087a41856/call-PESRCollection/PESRCollection/066417e3-3848-4dc7-b46e-173fdf2f9ae7/call-RunPESRCollection/cacheCopy/NA21133.disc.txt.gz.tbi"
     ],
+    "PE_files_list": "gs://gatk-sv-ref-panel-1kg/outputs/GATKSVPipelineBatch/38c65ca4-2a07-4805-86b6-214696075fef/PE_files_list.txt",
```
Comment (mwalker174) [inline inputs/values/ref_panel_1kg.json:318]: Seems like we will need to eliminate the array inputs to avoid redundancy / conflicts with the file lists. Can you file a ticket for that?

File: scripts/inputs/create_test_batch.py [hunk 2]
```diff
@@ -38,17 +54,47 @@ def replace_output_dir(value, execution_bucket, outputs_dir):
...
+
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("metadata", help="GATKSVPipelineBatch metadata JSON file")
     parser.add_argument("--final-workflow-outputs-dir", help="If used, final_workflow_outputs_dir "
                                                              "option from Cromwell config file")
     parser.add_argument("--execution-bucket", help="Cromwell execution bucket, required if "
                                                    "using --final-workflow-outputs-dir")
+    parser.add_argument("--file-list-bucket", help="Bucket to which to upload file lists")
```
Comment (mwalker174) [inline scripts/inputs/create_test_batch.py:92]: What do you think about setting a default for this to `--final-workflow-outputs-dir`?

**Review by mwalker174 — APPROVED** [review]
Thanks! Minor comments

File: inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/output_configurations/GatherBatchEvidenceOutputs.json.tmpl [hunk 3]
```diff
@@ -30,6 +30,7 @@
   "GatherBatchEvidence.std_delly_vcf": "${this.std_delly_vcf}",
   "GatherBatchEvidence.std_manta_vcf": "${this.std_manta_vcf}",
   "GatherBatchEvidence.std_melt_vcf": "${this.std_melt_vcf}",
+  "GatherBatchEvidence.std_scramble_vcf": "${this.std_scramble_vcf}",
```
Comment (cwhelan) [inline inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/output_configurations/GatherBatchEvidenceOutputs.json.tmpl:33]: This seems like an unrelated change not mentioned in the PR description -- just wanted to make sure it's intentionally added here.

File: wdl/GATKSVPipelineSingleSample.wdl [hunk 4]
```diff
@@ -262,10 +262,10 @@ workflow GATKSVPipelineSingleSample {
     RuntimeAttr? runtime_attr_depth_merge_pre_clusterbatch
 
     # Reference panel standardized caller VCFs
-    Array[File] ref_std_manta_vcfs
-    Array[File] ref_std_wham_vcfs
-    Array[File]? ref_std_melt_vcfs
-    Array[File]? ref_std_scramble_vcfs
+    File ref_std_manta_vcfs  # list of files, one per line
```
Comment (cwhelan) [inline wdl/GATKSVPipelineSingleSample.wdl:265]: Do you think we should change these input names to eg `ref_std_manta_vcf_list` for consistency's sake? I'm not hung up on it but usually when I see an input name that's pluralized I assume it's going to be an array.

**Review by cwhelan — APPROVED** [review]
Looks good, minor comments for your consideration.

File: inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/output_configurations/GatherBatchEvidenceOutputs.json.tmpl (hunk 3 above)
Comment (epiercehoffman) [reply inputs/templates/terra_workspaces/cohort_mode/workflow_configurations/output_configurations/GatherBatchEvidenceOutputs.json.tmpl:33]: Yeah I just noticed it while I was working on this branch and didn't want to make a whole separate PR for this one line

File: scripts/inputs/create_test_batch.py (hunk 2 above)
Comment (epiercehoffman) [reply scripts/inputs/create_test_batch.py:92]: I thought about this but when I ran it I used `gs://gatk-sv-ref-panel-1kg/outputs/GATKSVPipelineBatch/38c65ca4-2a07-4805-86b6-214696075fef/` as opposed to `gs://gatk-sv-ref-panel-1kg/outputs/` because I wanted it to be tied to the workflow that produced the outputs, and I didn't want it to be overwritten next time (so that old versions are still valid). I could extract the workflow subdirectory from the files, but wasn't sure that was worth the effort

File: inputs/values/ref_panel_1kg.json (hunk 1 above)
Comment (epiercehoffman) [reply inputs/values/ref_panel_1kg.json:318]: I'm not sure we want to replace Array[File] inputs with File inputs for pipeline intermediates because in Terra, those intermediate arrays get written into the data table and used that way, and requiring them to be in a File would mess up that flow. We could consider providing a File? and Array[File]? option every time, but that seems cumbersome. I thought it made more sense to limit the scope to pipeline _inputs_, even though it means some redundancy

File: wdl/GATKSVPipelineSingleSample.wdl (hunk 4 above)
Comment (epiercehoffman) [reply wdl/GATKSVPipelineSingleSample.wdl:265]: Fair enough

Comment (epiercehoffman) [reply wdl/GATKSVPipelineSingleSample.wdl]: Updated & re-validated WDLs and JSONs with womtool & Terra validation script
